pyccel/codegen/printing/ccode.py

```diff
@@ -822,13 +822,47 @@ def _print_PythonRound(self, expr):
     def _print_PythonMinMax(self, expr):
         arg = expr.args[0]
         primitive_type = arg.dtype.primitive_type
+        variadic_args = False
+        can_compare = True
 
-        if primitive_type is PrimitiveFloatingPointType() and len(arg) == 2:
+        if isinstance(primitive_type, (PrimitiveFloatingPointType, PrimitiveIntegerType)):
+            variadic_args = True
+
+        if primitive_type is PrimitiveComplexType():
+            can_compare = False
+
+        if isinstance(arg, Variable) and isinstance(arg.class_type , HomogeneousTupleType):
+            arg = PythonTuple(*arg)
+
+        if isinstance(arg, (PythonTuple, PythonList)) and variadic_args:
+            key = self.get_c_type(arg.class_type.element_type)
+            self.add_import(Import('stc/common', AsName(VariableTypeAnnotation(arg.dtype), key)))
+            return  f'{key}_{expr.name}({len(arg.args)}, {", ".join(self._print(a) for a in arg.args)})'
```

Comment:
I find this string quite hard to read. Maybe break it up slightly:
```suggestion
            args_code = ", ".join(self._print(a) for a in arg.args)
            return  f'{key}_{expr.name}({len(arg.args)}, {args_code})'
```

Reply:
fixed!